mlx_graphs/datasets/lanl_dataset.py

```diff
@@ -0,0 +1,320 @@
+        large `auth.txt.gz` file occurs, and csv files are written to
+        ``self.raw_path``.
+        """
+        if self.process_original_files:
+            print("Processing all authentication logs...")
+            split(
+                auth_file=os.path.join(self.original_path, self.auth_file),
+                redteam_file=os.path.join(self.original_path, self.redteam_file),
+                dst_path=self.raw_path,
+                graph_path_at_index_fn=self.graph_path_at_index,
+            )
+
+            print("Compressing all files to .gz...")
+            compress_and_remove_files(
+                folder_path=self.raw_path,
+                file_extension="csv",
+            )
+
+    def load_lazily(self, file_path: str, as_pandas_df: bool = False) -> GraphData:
+        if not os.path.isfile(file_path):
+            raise FileNotFoundError(f"Warning: file not found: {file_path}")
+
+        # If gzip compression, we read the csv while decompressing it.
+        compression = "gzip" if self._use_gzip else "infer"
+
+        # Reads the csv and assign an int to each column index.
+        df = pd.read_csv(
+            file_path,
+            index_col=False,
+            header=None,
+            names=range(LANL_TS, LANL_DST_USER + 1),
+            dtype={
+                LANL_TS: str,
+                LANL_SRC: int,
+                LANL_DST: int,
+                LANL_LABEL: int,
+                LANL_SUCCESS: int,
+                LANL_SRC_USER_TYPE: int,
+                LANL_SRC_USER: str,
+                LANL_DST_USER: str,
+            },
+            compression=compression,
+        )
+
+        # Additional preprocessing, normalisation, ...
+        df = self._preprocess_one_snapshot(df)
+
+        # Output schema.
+        df_adj = df[
+            [LANL_SRC, LANL_DST, LANL_TS, LANL_LABEL, LANL_SRC_USER, LANL_DST_USER]
+        ]
+        edge_feats = df[[LANL_SUCCESS, LANL_SRC_USER_TYPE]].to_numpy()
+
+        # If called from a loader, we usually return the df for further preprocessing
+        if as_pandas_df:
+            return (
+                df_adj,
+                edge_feats,
+            )
+
+        # By default, the dataset returns a GraphData with node features
+        graph = self.to_graphdata(
+            df_adj,
+            edge_feats,
+        )
+        graph = self.add_one_hot_node_features(graph)
+        return graph
+
+    def __concat__(self, items: list[GraphData]) -> GraphData:
+        return GraphData(
+            edge_index=mx.concatenate([g.edge_index for g in items], axis=1),
+            edge_labels=mx.concatenate([g.edge_labels for g in items], axis=0),
+            edge_features=mx.concatenate([g.edge_features for g in items], axis=0),
+            edge_timestamps=mx.concatenate([g.edge_timestamps for g in items], axis=0),
+            node_features=items[0].node_features,
+        )
+
+    def to_graphdata(self, df_adj: pd.DataFrame, edge_feats: np.array) -> GraphData:
```

Comment:
make private?

Reply:
No, it's used in `large_cybersecurity_loader.py`